index.bs

```diff
@@ -167,7 +170,9 @@ The <dfn attribute for="XRInputSource">gamepad</dfn> attribute is a {{Gamepad}}
  - More than one button
  - One or more axes
 
-Note: {{XRInputSource}}s in an {{XRSession}}'s {{XRSession/inputSources}} array are "live". As such values within them, such as the {{gamepad}} button and axis state, are updated in-place. This means that it doesn't work to save a reference to an {{XRInputSource}}'s {{gamepad}} on one frame and compare it to the same {{XRInputSource}}'s {{gamepad}} from a subsequent frame to test for state changes, because they will be the same object. Therefore developers that wish to compare input state from frame to frame should copy the state in question.
+The {{XRInputSource/gamepad}} object is "live", and any internal state is to be updated in-place every frame. This means that whenever an {{XRFrame}} is created, all {{XRInputSource/gamepad}}s attached to the {{XRFrame/session}} should have their state updated to whatever the corresponding gamepad state was at the time of creation of the pose and input state contained within the {{XRFrame}}. This should happen for {{XRFrame}}s created whilst processing the [=primary action=], not just [=XR animation frames=].
```

Comment:
Is there an exact algorithm we can link to in the webxr spec?

Reply:
There's no single algorithm. I could add an exported definition, but it's basically "process an XR AF" + each of the input event algorithms. It's going to expand as we add squeeze events. So I felt the reasonable thing to do would be to do it whenever XRFrame objects are created.

Reply:
Ah, I didn't mean a *single* algorithm.  We have algorithms defined for firing the events and for the frame loop.  What I meant to suggest was referencing them directly by name and explicitly stating at which step the data should be updated.  Perhaps we should tweak those algorithms to say something like "and here's where input sources and all their child attribute data is updated".

Reply:
We could, though it's weird when phrased that way because the core spec doesn't contain anything that needs updating on that tick. But I could work something into the core spec to reference here.

Reply:
Took a stab at it in https://github.com/immersive-web/webxr/pull/897 . Does that look about right?

Once we merge that we can add logic here that says that updating gamepad state is a frame update.

One question: should we have logic (either here or in that PR) to only update state so that it is newer? i.e. if a rAF callback with newer data happens before an older input event, should we be ignoring the data from input? We could also specify that this should never happen, but that's tricky.